statistical_disclosure_control/cell_key_perturbation/cell_key_perturbation_specification.md

```diff
@@ -0,0 +1,183 @@
+variable/characteristic.
+
+* The microdata needs a column containing record key, random numbers between
+0-255.
+
+* The method needs a ptable file, containing the perturbation/noise to be
+added.
+
+* The function call requires a list of variables to be tabulated
+
+* The record key needs to be numeric (integer) so it can be used in
+calculations (modulo sum). The other variables are expected to be categorical
+(they can be numeric but categorical is more suitable for frequency tables).
+
+## 5.2 Output records  
+
+---
+
+<!-- markdownlint-disable -->
+ |Variable definition| Type of variable| Format of specific variable (if applicable)| Expected range of the values| Meaning of the values | Expected level of aggregation | Comments |
+ |:---       |:---     |:---     |:---   |:--- |:--- |:--- |
+ |ckey       | integer || 0-255   | No intrinsic meaning. This value determines the perturbation to be added and ensures consistency | One value per cell in a frequency table | | |
+ |pcv        | integer || 0-750 | No intrinsic meaning. This value is an edited version of the count designed to reduce the necessary size of the ptable file. For cell values 0-750, pcv is the same as cell value | One value per cell in a frequency table | | |     
+ |count     | integer  || >=0  | The cell value after perturbation. The 'count' of how many records contain a combination of characteristics, plus the noise added by perturbation. E.g. there could be 14 males aged 18-24 in an area. Noise of +1 is applied to the cell so the post perturbation 'count' is 15. | One value per cell in a frequency table | | | 
+<!-- markdownlint-enable -->
+
+---
+
+* The count table will be in 'long' format sometimes known as tidy format, with
+one row per combination of characteristics. One column is included for each
+variable tabulated, e.g.: Local Authority, Age, Sex, count, pcv, pvalue, count;
+Bristol, 16-24, 1, 503, 503, +2, 505.
+
+## 5.3 Error handling  
+
+---
+
+Input data will need to be checked for errors, and that data contains record
+keys in the appropriate format.
+
+Any planned data editing should also be performed before the perturbation. The
+method will assume the data is correct and produce outputs accordingly.
+
+Where data contain missing values, the method will by default carry out
+perturbation and produce results for missing categories. If categories are to
+be combined or values imputed to replace missing data, this should be done
+before the perturbation.
+
+## 6.0 Overall Method  
+
+---
+
+The code produces a frequency table, counting how many records contain
+combinations of each of the supplied variables. In the same step, it takes the
+sum of the record keys of those records and takes modulo 256. The modulo is
+used so that cell keys are uniform in the range 0-255 (or 0-4095), which allows
+the method to specify and control the perturbation rate.
+
+To prevent the need for a very large ptable file, the rows 501-750 are also
```

Comment:
I don't really know what this means (rows 501-750) - is this referring to some pre-canned dataset? Remember this documentation will be public so (theoretically) this should be understandable to the target audience (which I am probably not) - if a data scientist/analyst would understand this then that is fine, but maybe it needs more clarity?